diesel/src/types/mod.rs

```diff
@@ -1,720 +1,109 @@
-//! Types which represent a SQL data type.
-//!
-//! The structs in this module are *only* used as markers to represent a SQL type.
-//! They should never be used in your structs.
-//! If you'd like to know the rust types which can be used for a given SQL type,
-//! see the documentation for that SQL type.
-//! Additional types may be provided by other crates.
-//!
-//! To see which SQL type can be used with a given Rust type,
-//! see the "Implementors" section of [`FromSql`].
-//!
-//! [`FromSql`]: trait.FromSql.html
-//!
-//! Any backend specific types are re-exported through this module
+// All items in this module are deprecated. They are rendered in docs.
+#![allow(missing_docs)]
+#![cfg(feature = "with-deprecated")]
+#![deprecated(since = "1.1.0", note = "Use `sql_types`, `serialize`, or `deserialize` instead")]
 
-pub mod ops;
-mod ord;
-#[macro_use]
-#[doc(hidden)]
-pub mod impls;
-mod fold;
+#[deprecated(since = "1.1.0", note = "Use `sql_types` instead")]
+pub use sql_types::*;
 
-use std::fmt;
-use std::ops::{Deref, DerefMut};
+#[deprecated(since = "1.1.0", note = "Use `deserialize` instead")]
+pub use deserialize::{FromSql, FromSqlRow};
 
-#[doc(hidden)]
-pub mod structs {
-    pub mod data_types {
-        //! Structs to represent the primitive equivalent of SQL types where
-        //! there is no existing Rust primitive, or where using it would be
-        //! confusing (such as date and time types). This module will re-export
-        //! all backend specific data structures when compiled against that
-        //! backend.
-        #[cfg(feature = "postgres")]
-        pub use pg::data_types::*;
-    }
-}
+#[deprecated(since = "1.1.0", note = "Use `serialize` instead")]
+pub use serialize::{IsNull, ToSql};
 
-pub use self::ord::SqlOrd;
-pub use self::fold::Foldable;
+#[deprecated(since = "1.1.0", note = "Use `sql_types::Bool` instead")]
+pub type Bool = ::sql_types::Bool;
```

Comment:
We reexport the whole `sql_types` module above so this should not be needed again?
(Same for all other types here)

Reply:
This is to work around bugs in Rust.